lametro/models.py

```diff
@@ -2,23 +2,56 @@
 import re
 from datetime import datetime, timedelta
 from dateutil.relativedelta import relativedelta
+import requests
 
 from django.conf import settings
 from django.db import models, connection
 from django.db.models.expressions import RawSQL
 from django.utils import timezone
 from django.contrib.auth.models import User
-from django.db.models import Max, Min, Prefetch, Case, When, Value
-
-import requests
+from django.db.models import Max, Min, Prefetch, Case, When, Value, Q
 
 from councilmatic_core.models import Bill, Event, Post, Person, Organization, \
     Action, EventMedia
 
 
 app_timezone = pytz.timezone(settings.TIME_ZONE)
 
+
+class LAMetroBillManager(models.Manager):
+    def get_queryset(self):
+        '''
+        The Councilmatic database contains both "private" and "public" bills. 
+        This issue thread explains why: 
+        https://github.com/datamade/la-metro-councilmatic/issues/345#issuecomment-455683240
+        
+        We do not display "private" bills in Councilmatic.
+        Metro staff devised three checks for knowing when to hide or show a report:
+
+        (1) Is the bill private (i.e., `restrict_view` is True)? Then, do not show it.
+        N.b., the scrapers contain logic for populating the restrict_view field:
+        https://github.com/opencivicdata/scrapers-us-municipal/blob/master/lametro/bills.py
+
+        (2) Does the Bill have a classification of "Board Box"? Then, show it.
+
+        (3) Is the Bill on a published agenda, i.e., an event with the 
+        status of "passed" or "cancelled"? Then, show it.
+
+        NOTE: Be sure to use LAMetroBill, rather than the base Bill class,
+        when getting bill querysets. Otherwise restricted view bills
+        may slip through the crevices of Councilmatic display logic.
+        '''
+        filtered_qs = super().get_queryset().exclude(restrict_view=True)\
+                                            .filter(Q(related_agenda_items__event__status='passed') | \
+                                                    Q(related_agenda_items__event__status='cancelled') | \
+                                                    Q(bill_type='Board Box'))\
+                                            .distinct()
```

Comment:
what's up with this distinct?

Reply:
A bill can appear on multiple event agendas, e.g., Bill 2019-0083 appears on the agenda for two events:

https://ocd.datamade.us/ocd-event/18f62517-80c2-4c09-9ee3-3d2a7486399d/
https://ocd.datamade.us/ocd-event/cbb92117-9eac-4d4b-a8a1-48299c3416a0/

The filters consider whether a bill has appeared on an Event with a particular status, meaning that the Django ORM runs a query like this:

```
SELECT DISTINCT "councilmatic_core_bill"."ocd_id", "councilmatic_core_bill"."ocd_created_at", "councilmatic_core_bill"."ocd_updated_at", "councilmatic_core_bill"."description", "councilmatic_core_bill"."identifier", "councilmatic_core_bill"."bill_type", "councilmatic_core_bill"."classification", "councilmatic_core_bill"."source_url", "councilmatic_core_bill"."source_note", "councilmatic_core_bill"."from_organization_id", "councilmatic_core_bill"."full_text", "councilmatic_core_bill"."html_text", "councilmatic_core_bill"."ocr_full_text", "councilmatic_core_bill"."abstract", "councilmatic_core_bill"."last_action_date", "councilmatic_core_bill"."updated_at", "councilmatic_core_bill"."legislative_session_id", "councilmatic_core_bill"."slug", "councilmatic_core_bill"."restrict_view" 
FROM "councilmatic_core_bill" 
LEFT OUTER JOIN "councilmatic_core_eventagendaitem" 
ON ("councilmatic_core_bill"."ocd_id" = "councilmatic_core_eventagendaitem"."bill_id") 
LEFT OUTER JOIN "councilmatic_core_event" 
ON ("councilmatic_core_eventagendaitem"."event_id" = "councilmatic_core_event"."ocd_id") WHERE (NOT ("councilmatic_core_bill"."restrict_view" = True) 
AND ("councilmatic_core_event"."status" = passed 
OR "councilmatic_core_event"."status" = cancelled 
OR "councilmatic_core_bill"."bill_type" = Board Box))
```

Without distinct, we'd get duplicate bills in our queryset.

Reply:
ah, that makes sense! can you add a brief comment to that effect in the manager?

Reply:
'tis done!